### ISSUE TITLE
Add events to UnitOfWork

### ISSUE THREAD

**Issue by AndrolGenhald** [graphaware/neo4j-php-ogm#13]
Implement lifecycle events similar to doctrine-orm: http://doctrine-orm.readthedocs.io/projects/doctrine-orm/en/latest/reference/events.html

Since much of the implementation resides in doctrine/common, I'm hoping I'll be able to get at least some of the events implemented this weekend without too much trouble.

**Comment by ikwattro** [comment]
Thanks, yeah this is planned of course but not really in the priorities for now, if you come up with a PR that would be wonderful.

**Comment by ikwattro** [comment]
@AndrolGenhald Do you have any progress on this, if yes please open a PR so I can contribute and make some improvements for this implementation.

**Comment by AndrolGenhald** [comment]
I haven't started anything yet, I spent last night looking through doctrine-orm's implementation of lifecycle events.

**Comment by ikwattro** [comment]
Right, I'll check also and maybe prepare some entry points for it. Thanks !

**Comment by ikwattro** [comment]
What is your actual use case for doctrine events !

**Comment by ikwattro** [comment]
Ha yeah, DoctrineModule Form element

**Comment by AndrolGenhald** [comment]
I'm mainly looking to subscribe to the onFlush event to create historical copies of entities when they are changed, right now I'm handling it everywhere I call flush.

**Comment by ikwattro** [comment]
Interesting ! Do you have a screenshot of what your graph looks like after the historic is created ? You can even share privately if you want : christophe at graphaware dot com

**Comment by AndrolGenhald** [comment]
Sequence of events:
- Add 'Entity'
- Add 'attribute 1'
- Add 'attribute 2'
- Rename 'Entity' -> 'Entity (changed)'
- Add 'attribute 3'
- Rename 'attribute 1' -> 'attribute 1 (changed)'
- Rename 'Entity' -> 'Entity (changed again)'
![screenshot](https://cloud.githubusercontent.com/assets/7794502/17263662/c7a11a92-55a7-11e6-988a-e7a1747bf364.png)

**Comment by ikwattro** [comment]
Very cool !

**Comment by AzeemMichael** [comment]
There seems to be no `recomputeSingleEntityChangeSet($meta, $entity);` in unit of work. Wanted to create event listener for `onFlush` so `User::updatedAt` field is set to current timestamp on node update.
